lib/active_merchant/billing/gateways/kushki.rb

```diff
@@ -265,8 +267,12 @@ def commit(action, params)
         )
       end
 
-      def ssl_invoke(action, params)
-        if %w[void refund].include?(action)
+      def ssl_invoke(action, params, options)
+        if %w[refund].include?(action) && options[:partial_refund] == true
+          # removes ticketNumber from request because gateway will reject if included in request body
+          post_params = params.except(:ticketNumber)
+          ssl_request(:delete, url(action, params), post_data(post_params), headers(action))
+        elsif %w[void refund].include?(action)
           ssl_request(:delete, url(action, params), nil, headers(action))
```

Comment:
I would suggest moving the top part to here. It could look something like this

elsif %w[void refund].include?(action)
   data = options[:partial_refund] == true ? post_data(params.except(:ticketNumber)) : nil
   ssl_request(:delete, url(action, params), data, headers(action))